hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeImpl.java

```diff
@@ -46,18 +47,22 @@ public class NodeImpl implements Node {
    * {@link NetConstants#PATH_SEPARATOR})
    * @param location this node's location
    */
-  public NodeImpl(String name, String location, int cost) {
-    if (name != null && name.contains(PATH_SEPARATOR_STR)) {
+  public NodeImpl(StringWithByteString name, StringWithByteString location, int cost) {
+    if (name != null && name.toString() != null && name.toString().contains(PATH_SEPARATOR_STR)) {
```

Comment:
Let's enforce that `StringWithByteString.toString()` is never null?  We may add `Objects.requireNonNull(..)` in the `StringWithByteString` constructor.
- If the original string is null, then `name == null`;
- otherwise,  `name != null && name.toString() != null`.